core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -97,13 +103,26 @@ case class ExceptionFailure(
     description: String,
     stackTrace: Array[StackTraceElement],
     fullStackTrace: String,
-    metrics: Option[TaskMetrics])
+    metrics: Option[TaskMetrics],
+    exceptionWrapper: Option[ThrowableSerializationWrapper])
```

Comment:
Would you mind adding a test for this guy? Could make a new Exception subtype which throws an exception in readObject() :)